packages/ui/src/hooks/use-social-sign-in-listener.ts

```diff
@@ -1,51 +1,72 @@
+import type { RequestErrorBody } from '@logto/schemas';
 import { SignInMode } from '@logto/schemas';
 import { useEffect, useCallback, useContext, useMemo } from 'react';
 import { useTranslation } from 'react-i18next';
-import { useParams, useNavigate } from 'react-router-dom';
+import { useNavigate } from 'react-router-dom';
+import { validate } from 'superstruct';
 
 import { signInWithSocial } from '@/apis/interaction';
+import { socialAccountNotExistErrorDataGuard } from '@/types/guard';
 import { parseQueryParameters } from '@/utils';
 import { stateValidation } from '@/utils/social-connectors';
 
 import type { ErrorHandlers } from './use-api';
 import useApi from './use-api';
 import { PageContext } from './use-page-context';
 import useRequiredProfileErrorHandler from './use-required-profile-error-handler';
+import { useSieMethods } from './use-sie';
```

Comment:
nit: maybe `useSIEMethods` since SIE is the abbreviation of `Sign-In Experience`

Reply:
We use Sie cross all the code bases, like core.